docs/resources/Auto_Moderation.md

```diff
@@ -0,0 +1,209 @@
+| exempt_roles     | array of snowflakes | the role ids that should not be affected by the rule (Maximum of 20)                                                       |
+| exempt_channels  | array of snowflakes | the channel ids that should not be affected by the rule (Maximum of 50)                                                    |
+
+###### Example Auto Moderation Rule
+
+```json
+{
+  "id": "969707018069872670",
+  "guild_id": "613425648685547541",
+  "name": "Keyword Filter 1",
+  "creator_id": "423457898095789043",
+  "trigger_type": 1,
+  "event_type": 1,
+  "actions": [
+    {
+      "type": 1,
+      "metadata": {}
+    },
+    {
+      "type": 2,
+      "metadata": { "channel_id": "123456789123456789" }
+    }
+  ],
+  "trigger_metadata": {
+    "keyword_filter": ["cat*", "*cat", "*ana*"]
+  },
+  "enabled": true,
+  "exempt_roles": ["323456789123456789", "423456789123456789"],
+  "exempt_channels": ["523456789123456789"],
+}
+```
+
+###### Trigger Types
+
+Characterizes what type of information will be checked to determine whether a rule is triggered.
+
+| Type                 | Value   | Description                                                           |
+| -------------------- | ------- | --------------------------------------------------------------------- |
+| KEYWORD              | 1       | check if content contains words from a user defined list of keywords  |
+| HARMFUL_LINK         | 2       | check if content contains any harmful links                           |
+| SPAM                 | 3       | check if content represents generic spam                              |
+| KEYWORD_PRESET       | 4       | check if content contains words from internal pre-determined wordsets |
+
+###### Trigger Metadata
+
+Additional data used to determine whether a rule should be triggered. Different fields are relevant based on the
+value of [trigger_type](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-rule-object-trigger-types).
+
+| Field          | Type                                                                                                | Associated Trigger Types      | Description                                                               | 
+| -------------- | --------------------------------------------------------------------------------------------------- | ----------------------------- | ------------------------------------------------------------------------- |
+| keyword_filter | array of strings                                                                                    | KEYWORD                       | substrings to match in content                                            |
+| keyword_lists  | array of [wordset types](#DOCS_RESOURCES_AUTO_MODERATION/auto-moderation-rule-object-wordset-types) | KEYWORD_PRESET                | additional metadata needed during execution for this specific action type |
```

Comment:
this description is very vague

Reply:
this is because I copy pasted this for a different section and forgot to change it 🤦‍♂️

Reply:
how does this sound?
```
keyword_filter -- words to try to match in content
keyword_lists -- which pre-defined lists of words to try to match in content
```
Also `keyword_filter` and especially `keyword_lists` are not good names. We will prob look at changing this before API is released